tools/ci_build/build.py

```diff
@@ -276,7 +276,7 @@ def convert_arg_line_to_args(self, arg_line):
 
     # Build options
     parser.add_argument(
-        "--cmake_extra_defines", nargs="+",
+        "--cmake_extra_defines", nargs="+", action='append',
```

Comment:
there is another action 'extend'. If we use it, we don't need to flatten the array in line 2040. See also [link](https://docs.python.org/3/library/argparse.html#action).

However, action 'extend' only works on Python 3.8+. For backward compatibility, I don't use it and do manual array flattening in this PR.